pandas/core/generic.py

```diff
@@ -663,21 +665,36 @@ def droplevel(self: FrameOrSeries, level, axis=0) -> FrameOrSeries:
         result = self.set_axis(new_labels, axis=axis, inplace=False)
         return result
 
-    def pop(self: FrameOrSeries, item) -> FrameOrSeries:
-        """
-        Return item and drop from frame. Raise KeyError if not found.
+    _shared_docs[
+        "pop"
+    ] = """
+        Return item and drops from %(klass)s. Raise KeyError if not found.
 
         Parameters
         ----------
-        item : str
-            Label of column to be popped.
+        item : %(axes_single_arg)s
+            %(pop_item_param_info)s
 
         Returns
         -------
-        Series
+        %(pop_return_type)s
 
         Examples
         --------
+        **Series**
```

Comment:
Rather than having a Series section and a DataFrame section, could you have the Series example just show up in the Series.pop docs, and the existing DataFrame example just show up in DataFrame.pop?

Reply:
I tried to make separate examples first. but didn't know how to differentiate docstrings from parent except _shared_doc_kwargs variable. 
Could you provide an example from where I can take reference.

Reply:
I think DataFrame and Series `.to_markdown` have different example, perhaps check those?

Reply:
Yes, it helped a lot. Thanks!